src/GraphQL/Types/Input/Substrate/Traits/HasSigningAccountField.php

```diff
@@ -30,10 +30,14 @@ public function getSigningAccountField(
     /**
      * Get the signing account.
      */
-    public function getSigningAccount(array $args)
+    public function getSigningAccount(array $args): ?Model
     {
-        return WalletService::firstOrStore([
-            'account' => empty($signing = Arr::get($args, 'signingAccount')) ? Account::daemonPublicKey() : $signing,
-        ]);
+        if (!empty($signing = Arr::get($args, 'signingAccount'))) {
+            return WalletService::firstOrStore([
```

Comment:
I think there's a problem here, how will the signingAccount be populated with multi-tenant? I don't think the wallet daemon knows what's the wallet address of the user

I think we need to fix this part in `Account.php` instead

```
public static function daemonPublicKey(): ?string
{
        return static::$publicKey ? SS58Address::getPublicKey(static::$publicKey) : null;
}
```

Take note that for multi-tenant we are setting this `static::$publicKey` attribute here https://github.com/enjin/platform-multi-tenant/blob/feature/PLA-845/multi-tenant-system/src/Services/Auth/Drivers/SanctumAuth.php#L37


```suggestion
       $account =  empty($signing = Arr::get($args, 'signingAccount')) ? Account::daemonPublicKey() : $signing
       if ($account) {
              return WalletService::firstOrStore(['account' => $account]);
       }
       return null;
```

Reply:
I'm not sure I got your point here. The signing account is an address, the address is unique. We don't have the same address for different users. Also each user has its own daemon, we don't need to know the user address.

Reply:
tagging @v16Studios as he is more used with the multi-tenant package.